tests/integration-tests/conftest.py

```diff
@@ -313,6 +317,62 @@ def _cluster_factory(cluster_config, extra_args=None, raise_on_error=True):
         factory.destroy_all_clusters(test_passed=request.node.rep_call.passed)
 
 
+@pytest.fixture(scope='session')
+def api_server_factory(cfn_stacks_factory, request, public_ecr_image_uri,
+                       api_definition_s3_uri, api_infrastructure_s3_uri):
+    """Creates a factory for deploying API servers on-demand to each region."""
+    api_servers = {}
+
+    def _api_server_factory(server_region):
+        api_stack_name = generate_stack_name("integ-tests-api", request.config.getoption("stackname_suffix"))
+
+        if not api_infrastructure_s3_uri:
+            stack_template_path = os.path.join("..", "..", "api", "infrastructure", "parallelcluster-api.yaml")
+            with open(stack_template_path) as stack_template_file:
+                stack_template_data = stack_template_file.read()
+
+        params = [{"ParameterKey": "PublicEcrImageUri", "ParameterValue": public_ecr_image_uri},
+                  {"ParameterKey": "ApiDefinitionS3Uri", "ParameterValue": api_definition_s3_uri}]
+
+        if server_region not in api_servers:
+            logging.info(f"Creating API Server stack: {api_stack_name} in {server_region}")
+            try:
+                set_credentials(server_region, request.config.getoption("credential"))
+                stack = CfnStack(name=api_stack_name,
+                                 region=server_region,
+                                 parameters=params,
+                                 capabilities=['CAPABILITY_IAM', 'CAPABILITY_AUTO_EXPAND'],
+                                 template=api_infrastructure_s3_uri or stack_template_data)
+                cfn_stacks_factory.create_stack(stack)
+            finally:
+                unset_credentials()
+            api_servers[server_region] = stack
+        else:
+            logging.info(f"Found cached API Server stack: {api_stack_name} in {server_region}")
+
+        return api_servers[server_region]
+
+    yield _api_server_factory
+
+
+@pytest.fixture(scope="class")
+def api_client(region, api_server_factory, api_uri):
+    """Define a fixture for an API client that interacts with the
+    pcluster api."""
+    from pcluster_client import Configuration, ApiClient
```

Comment:
how is this import working?are you manually modifying your pythonpath to add the client dir?
I'm wondering if it makes sense to add the relative path to the client source in the requirements.txt defined under tests/integration-tests. The downside could be that `pip install -r requirements` will only work from within the tests/integration-tests dir.

Reply:
I did a `pip install -e .` on the generated code...

_something_ needs to happen but I wasn't sure if it's jenkins as a separate external step or embedded in req's (or something else)?